src/todo/UI.lua

```diff
@@ -121,40 +135,47 @@ function todo.create_add_edit_frame(player)
 
     local table = frame.add({
         type = "table",
+        style = "todo_table_default",
         name = "todo_add_task_table",
         column_count = 2
     })
 
     table.add({
         type = "label",
+        style = "todo_label_default",
         name = "todo_add_task_label",
         caption = {"todo.add_task"}
     })
 
-    local textbox = table.add({
+    table.add({
         type = "text-box",
-        name = "todo_new_task_textbox"
+        style = "todo_textbox_default",
+        name = "todo_new_task_textbox",
+        text = task and task.task or ""
     })
-    textbox.style.minimal_width = 300
-    textbox.style.minimal_height = 100
 
     table.add({
         type = "label",
+        style = "todo_label_default",
         name = "todo_add_assignee_label",
         caption = {"todo.add_assignee"}
     })
 
-
-    local players, _, c = todo.get_player_list()
-    local index = 1
-    if( todo.is_auto_assign(player) and c == 1 ) then
-        index = 2
+    local players, lookup, c = todo.get_player_list()
+    local assign_index = 1
+    if task then
+        assign_index = task.assignee and lookup[task.assignee] or 1
+    else
+        if( todo.is_auto_assign(player) and c == 1 ) then
```

Comment:
Hmm, would it be better to merge this with the `else` to form an `elseif`?

Reply:
yes